app/controllers/prove_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ProveController < ApplicationController
+  before_filter :check_write_permissions
+
+  def create
```

Comment:
Reviewer: I don't see a problem here. If this really is too complex, please object and I'll change it.